src/app/dashboard/dashboard.component.ts

```diff
@@ -40,9 +40,9 @@ export class DashboardComponent implements OnInit {
   }
 
   ngOnInit() {
-    console.log('test1' + this.apiService.getBalance());
+    console.log('test1' + this.apiService.getBalanceToThisDay());
     return this.apiService
```

Comment:
Nie ma takiej potrzeby żeby dodawać tu returna, wystarczy:
this.apiService.getBalanceToThisDay().subscribe(data => this.balanceData = data);

Return powoduje wyjście z funkcji po wykonaniu.  Dodanie go w ngOnInit spowoduje, że po wykonaniu funkcji z apiService - lifecycle hook ngOnInit zakończy się. W tym przypadku nie potrzebujemy tego, ponieważ nie wywołujemy żadnych innych metod i wyjdziemy z ngOnInit tak czy inaczej.